client/my-sites/site-settings/seo-settings/form.jsx

```diff
@@ -466,17 +469,17 @@ export const SeoForm = React.createClass( {
 					path="/settings/seo/:site"
 					title="Site Settings > SEO"
 				/>
-				{ isSitePrivate && hasBusinessPlan( site.plan ) &&
+				{ ( isSitePrivate || isSiteHidden ) && hasBusinessPlan( site.plan ) &&
 					<Notice
 						status="is-warning"
 						showDismiss={ false }
-						text={ translate(
-							'SEO settings are disabled because the ' +
-							'site visibility is not set to Public.'
-						) }
+						text={ isSitePrivate
+							? translate( "SEO settings aren't recognized by search engines while your site is Private." )
+							: translate( "SEO settings aren't recognized by search engines while your site is Hidden." )
+						}
 					>
 						<NoticeAction href={ generalTabUrl }>
-							{ translate( 'View Settings' ) }
+							{ translate( 'View privacy settings' ) }
```

Comment:
Nice suggestion, personally I'd change this to "Privacy Settings", it would also help with keeping everything on one line.